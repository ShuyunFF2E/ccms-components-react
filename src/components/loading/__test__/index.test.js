import React, { Component } from 'react';
import { render, mount } from 'enzyme';
import mountTest from '@tests/shared/mountTest';
import { prefixCls } from '@utils';
import Loading from '../index';
import Toggle from '../../toggle';

const classSelector = `${prefixCls}-loading`;

describe('Loading', () => {
	mountTest(Loading);

	it('renders correctly', () => {
		const wrapper = render(<Loading />);
		expect(wrapper).toMatchSnapshot();
	});

	it('renders different size', () => {
		expect(render(<Loading size="small" />)).toMatchSnapshot();
		expect(render(<Loading />)).toMatchSnapshot();
		expect(render(<Loading size="large" />)).toMatchSnapshot();
	});

	it('should render correctly with loading state', () => {
		const wrapper = mount(<Loading />);
		expect(wrapper.find(`.${classSelector}`)).toHaveLength(1);
	});

	it('should render text with tip', () => {
		const tip = 'Loading加载提示';
		const wrapper = render(<Loading tip={tip} />);
		expect(wrapper.find('.cloud-loading-text').text()).toEqual(tip);
	});

	it('should render correctly after delay time', () => {
		class TestComponent extends Component {
			constructor(props) {
				super(props);
				this.state = {
					loading: false
				};
			}

			handleChange = loading => {
				this.setState({ loading });
			};

			render() {
				return (
					<div>
						<Toggle checked={this.state.loading} onChange={this.handleChange} />
						<Loading delay={1000} loading={this.state.loading} />
					</div>
				);
			}
		}

		const wrapper = mount(<TestComponent />);

		jest.useFakeTimers();

		expect(wrapper.find('Loading').state('delayShow')).toBeFalsy();

		wrapper.find(Toggle).simulate('click');

		jest.advanceTimersByTime(1000);

		expect(wrapper.find('Loading').state('delayShow')).toBeTruthy();

		jest.useRealTimers();
	});

	it('should render correctly has children props', () => {
		const wrapper = mount(
			<Loading>
				<div className="test">这里是我要秀的文案啦。</div>
			</Loading>
		);
		expect(wrapper.find('.test')).toHaveLength(1);
	});
});
