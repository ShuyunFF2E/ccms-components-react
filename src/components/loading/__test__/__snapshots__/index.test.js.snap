// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `
<div
  class="cloud-loading"
>
  <div
    class="cloud-loading-container"
  >
    <div
      class="cloud-loading-animation cloud-loading-animation-default"
      style="width:12px;height:16px"
    >
      <span
        style="width:2px"
      />
      <span
        style="width:2px"
      />
      <span
        style="width:2px"
      />
    </div>
  </div>
</div>
`;

exports[`Loading renders different size 1`] = `
<div
  class="cloud-loading"
>
  <div
    class="cloud-loading-container"
  >
    <div
      class="cloud-loading-animation cloud-loading-animation-small"
      style="width:12px;height:10px"
    >
      <span
        style="width:2px"
      />
      <span
        style="width:2px"
      />
      <span
        style="width:2px"
      />
    </div>
  </div>
</div>
`;

exports[`Loading renders different size 2`] = `
<div
  class="cloud-loading"
>
  <div
    class="cloud-loading-container"
  >
    <div
      class="cloud-loading-animation cloud-loading-animation-default"
      style="width:12px;height:16px"
    >
      <span
        style="width:2px"
      />
      <span
        style="width:2px"
      />
      <span
        style="width:2px"
      />
    </div>
  </div>
</div>
`;

exports[`Loading renders different size 3`] = `
<div
  class="cloud-loading"
>
  <div
    class="cloud-loading-container"
  >
    <div
      class="cloud-loading-animation cloud-loading-animation-large"
      style="width:18px;height:22px"
    >
      <span
        style="width:3px"
      />
      <span
        style="width:3px"
      />
      <span
        style="width:3px"
      />
    </div>
  </div>
</div>
`;
